src/core/public/loading_count/loading_count_service.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import * as Rx from 'rxjs';
+import { pairwise, startWith } from 'rxjs/operators';
+
+import { FatalErrorsStartContract } from '../fatal_errors';
+
+interface Deps {
+  fatalErrors: FatalErrorsStartContract;
+}
+
+export class LoadingCountService {
+  private readonly count$ = new Rx.BehaviorSubject(0);
+  private readonly subscriptions: Rx.Subscription[] = [];
+
+  public start({ fatalErrors }: Deps) {
+    return {
+      add: (count$: Rx.Observable<number>) => {
+        this.subscriptions.push(
+          count$
+            .pipe(
+              startWith(0),
+              pairwise()
+            )
+            .subscribe({
+              next: ([prev, next]) => {
+                const delta = next - prev;
+                this.count$.next(this.count$.getValue() + delta);
```

Comment:
__issue:__ it can easily go to negative numbers, what should do we in this case? Call `fatalErrors.add` or just log this weird behavior and keep `0` as the minimum count?

Reply:
__UPDATE:__ just put `console.log` here and indeed I've seen `-1` for `this.count$.getValue() + delta` a couple of times (with x-pack plugins).

Reply:
Hmm, we currently only have two things that are contributing to the loading count: the length of the `$http.pendingRequest` array (can't ever go below zero) and calls to `chrome.loadingCount.increment()` and `decrement()` from here: https://github.com/elastic/kibana/blob/f0cc432fafd6f7053f32d493aba8b576dcb20539/src/ui/ui_settings/public/send_request.js#L24

Based on that I can't imagine how we are getting negative loading counts...

Reply:
Can you try running in the debugger and tracing back up the call stack to see what is causing the loading count to go negative?

Reply:
Can't reproduce it with the latest changes, so it's all good now! :)